lib/spack/spack/hooks/autopush.py

```diff
@@ -0,0 +1,39 @@
+# Copyright 2013-2024 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+import argparse
+
+import spack.mirror
+from spack.cmd import buildcache as bc
+
+
+def post_install_in_db(spec):
+    pkg = spec.package
+    # Package was installed from source
+    if not pkg.installed_from_binary_cache:
+        hash = "/" + spec.dag_hash()
+        # Iterate over all mirrors with autopush==True
+        all_mirrors = spack.mirror.MirrorCollection(binary=True)
+        autopush_mirrors = [mirror for mirror in all_mirrors.values() if mirror.get_autopush()]
+        for mirror in autopush_mirrors:
+            # Create and execute buildcache push command
+            bc_parser = argparse.ArgumentParser()
```

Comment:
Can you use

```spack.binary_distribution.push_or_raise```

instead of `spack.cmd.buildcache.push_fn`?

It doesn't support OCI stuff, but that's fine, cause that doesn't work with `--only=package` anyways.